### PR TITLE
feat: added forced deploy

### LINKED ISSUE #62: Viya deployment when target folder already exists

### PR DESCRIPTION
**Pull request by YuryShkoda**
closes #62

### CODE REVIEW

**Comment by allanbowe** [comment]
This is great except one thing - we should have a generic function for deleting (or, recycling) folders, as it's something we can re-use in other apps.

**Review by allanbowe — CHANGES_REQUESTED** [review]

File: src/SASViyaApiClient.ts [hunk 1]
```diff
@@ -1085,6 +1121,91 @@ export class SASViyaApiClient {
...
+      return { result: null }
+    })
+
+    if (!folder) return undefined
+
+    return folder
+  }
+
+  /**
+   * Moves folder to the recycle bin
```
[the remark this answers is not in the document]
Comment (YuryShkoda) [reply src/SASViyaApiClient.ts]: done with a few changes

**Review by medjedovicm — DISMISSED** [review]

**Comment by krishna-acondy** [comment]
@YuryShkoda It might be a good idea to add doc comments to any methods you changed in `SASjs.ts` - it'll save us time later on.

File: src/SASViyaApiClient.ts [hunk 2]
```diff
@@ -470,7 +472,8 @@ export class SASViyaApiClient {
     folderName: string,
     parentFolderPath?: string,
     parentFolderUri?: string,
-    accessToken?: string
+    accessToken?: string,
+    isForced?: boolean
```
Comment (allanbowe) [inline src/SASViyaApiClient.ts:476]: this should be in the typeDoc signature

**Review by allanbowe — CHANGES_REQUESTED** [review]

File: src/SASViyaApiClient.ts (hunk 2 above)
Comment (YuryShkoda) [reply src/SASViyaApiClient.ts:478]: fixed

File: src/SASViyaApiClient.ts [hunk 3]
```diff
@@ -499,6 +506,35 @@ export class SASViyaApiClient {
           accessToken
         )
         console.log(`Parent Folder "${newFolderName}" successfully created.`)
+        parentFolderUri = `/folders/folders/${parentFolder.id}`
+      } else if (isForced && accessToken && !this.isForceDeploy) {
+        this.isForceDeploy = true
+
+        await this.deleteFolder(parentFolderPath, accessToken)
```
Comment (saadjutt01) [inline src/SASViyaApiClient.ts:513]: need to have condition here, if unable to deleteFolder for any reason. 
we can show error cannot move/delete current folder.

Comment (YuryShkoda) [reply src/SASViyaApiClient.ts:513]: improved error handling in moveFolder function

**Review by allanbowe — APPROVED** [review]

**Pull request merged**